src/UI/Areas/Identity/Pages/V4/Account/Register.cshtml.cs

```diff
@@ -82,10 +80,8 @@ public override async Task<IActionResult> OnPostAsync(string returnUrl = null)
             returnUrl = returnUrl ?? Url.Content("~/");
             if (ModelState.IsValid)
             {
-                var user = CreateUser();
-
-                await _userStore.SetUserNameAsync(user, Input.Email, CancellationToken.None);
```

Comment:
Is the idea that any custom data that a custom `IUserStore` is pushing onto a a `TUser` instance needs to be explicitly added to the app's pages? i.e. we can't make it "automatic"?

Reply:
Yeah basically its possible they would want to set arbitrary data from the register page into the initial user before creating it.

Age, Gender, SSN, etc...  

The factory in this PR is an abstract class instead of an interface so we can add something like
`Create(IDictionary<string, string> registerData)` in the future if we wanted without a breaking change